Code/Legacy/CryCommon/CryAssert_Linux.h

```diff
@@ -72,15 +72,15 @@ bool CryAssert(const char* szCondition, const char* szFile, unsigned int line, b
 
     static const int max_len = 4096;
     static char gs_command_str[4096];
-    static CryLockT<CRYLOCK_RECURSIVE> lock;
+    static AZStd::recursive_mutex lock;
 
     gEnv->pSystem->OnAssert(szCondition, gs_szMessage, szFile, line);
 
     size_t file_len = strlen(szFile);
 
     if (!gEnv->bNoAssertDialog && !gEnv->bIgnoreAllAsserts)
     {
-        CryAutoLock< CryLockT<CRYLOCK_RECURSIVE> > lk (lock);
+        AZStd::lock_guard<AZStd::recursive_mutex> lk (lock);
```

Comment:
Recommendation: The lock_guard can be replaced with a scoped lock

```suggestion
        AZStd::scoped_lock lk(lock);
```